src/current/v25.2/row-level-security.md

```diff
@@ -0,0 +1,561 @@
+
+For example, imagine a SaaS application serving multiple tenants, with all invoice data residing in a single `invoices` table. This table is distinguished by a `tenant_id` column. The application ensures that each user session is associated with a specific `tenant_id`.
+
+#### Create multi-tenant schema
+
+First, create the schema and index for the `tenants` and `invoices` tables. Next, add an index on `tenant_id` for increased lookup performance.
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+CREATE TABLE IF NOT EXISTS tenants (
+    tenant_id UUID PRIMARY KEY DEFAULT gen_random_uuid(),
+    name TEXT NOT NULL UNIQUE
+);
+
+CREATE TABLE IF NOT EXISTS invoices (
+    invoice_id SERIAL PRIMARY KEY,
+    tenant_id UUID NOT NULL REFERENCES tenants(tenant_id),
+    customer_name TEXT NOT NULL,
+    amount NUMERIC(12, 2) NOT NULL,
+    created_at TIMESTAMPTZ DEFAULT now()
+);
+
+CREATE INDEX idx_invoices_tenant_id ON invoices(tenant_id);
+~~~
+
+Populate the schema with data:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+-- Insert the known tenants first to satisfy FK constraints
+INSERT INTO tenants (tenant_id, name) VALUES
+    ('9607a12c-3c2f-407b-ae3c-af903542395b', 'Tenant A Inc.'),
+    ('8177c2fc-3b55-47b7-bf84-38bd3a3e9c0a', 'Tenant B Solutions');
+
+-- Insert some additional dummy tenants
+INSERT INTO tenants (name) VALUES
+    ('Example Corp'),
+    ('Global Widgets Ltd.');
+
+-- Now, populate the invoices table
+-- Use the known tenant UUIDs and some generated ones (assuming default gen_random_uuid works or select from tenants)
+
+-- Invoice for Tenant A
+INSERT INTO invoices (tenant_id, customer_name, amount) VALUES
+    ('9607a12c-3c2f-407b-ae3c-af903542395b', 'Customer One', 1500.75);
+
+-- Invoice for Tenant B
+INSERT INTO invoices (tenant_id, customer_name, amount) VALUES
+    ('8177c2fc-3b55-47b7-bf84-38bd3a3e9c0a', 'Customer Two', 899.00);
+
+-- Another invoice for Tenant A
+INSERT INTO invoices (tenant_id, customer_name, amount) VALUES
+    ('9607a12c-3c2f-407b-ae3c-af903542395b', 'Customer Three', 210.50);
+
+-- Invoice for Example Corp (assuming its UUID was generated)
+INSERT INTO invoices (tenant_id, customer_name, amount)
+    SELECT tenant_id, 'Customer Four', 5000.00
+    FROM tenants WHERE name = 'Example Corp';
+
+-- Invoice for Global Widgets Ltd. (assuming its UUID was generated)
+INSERT INTO invoices (tenant_id, customer_name, amount)
+    SELECT tenant_id, 'Customer Five', 120.99
+    FROM tenants WHERE name = 'Global Widgets Ltd.';
+
+-- Yet another invoice for Tenant B
+INSERT INTO invoices (tenant_id, customer_name, amount) VALUES
+    ('8177c2fc-3b55-47b7-bf84-38bd3a3e9c0a', 'Customer Six', 345.67);
+~~~
+
+#### Define user roles for app developer
+
```

Comment:
Probably needs a line of descriptive text

Reply:
updated